src/views/resultsProviders.ts

```diff
@@ -1,196 +1,222 @@
-		map: Map<string, vscode.Diagnostic[]>
-	) {
-		if (!folder) {
-			return;
-		}
-		const filePath = vscode.Uri.joinPath(folder?.uri, node.fileName).toString();
-		// Needed because vscode uses zero based line number
-		const column = node.column > 0 ? +node.column - 1 : 1;
-		const line = node.line > 0 ? +node.line - 1 : 1;
-		const length = column + node.length;
-		const startPosition = new vscode.Position(line, column);
-		const endPosition = new vscode.Position(line, length);
-		const range = new vscode.Range(startPosition, endPosition);
-
-		const diagnostic = new vscode.Diagnostic(range, label, severity);
-		if (map.has(filePath)) {
-			map.get(filePath)?.push(diagnostic);
-		} else {
-			map.set(filePath, [diagnostic]);
-		}
-	}
-
-	private reduceGroups(
-		obj: AstResult,
-		previousValue: TreeItem,
-		currentValue: string
-	) {
-		let value = getProperty(obj, currentValue);
-
-		// Needed to group by filename in kics, in case nothing is found then its a kics result and must be found inside data.filename
-		if (currentValue === GroupBy.fileName && value.length === 0) {
-			value = getProperty(obj.data, GroupBy.fileName.toLowerCase());
-		}
-
-		if (!value) {
-			return previousValue;
-		}
-
-		const tree = previousValue.children
-			? previousValue.children.find(
-				(item) => item.label === value.replaceAll("_", " ")
-			)
-			: undefined;
-		if (tree) {
-			tree.setDescription();
-			return tree;
-		}
-
-		const newTree = new TreeItem(
-			value.replaceAll("_", " "),
-			undefined,
-			undefined,
-			[]
-		);
-		previousValue.children?.push(newTree);
-		return newTree;
-	}
-
-}
\ No newline at end of file
+  protected _onDidChangeTreeData: EventEmitter<TreeItem | undefined> =
+    new EventEmitter<TreeItem | undefined>();
+  readonly onDidChangeTreeData: vscode.Event<TreeItem | undefined> =
+    this._onDidChangeTreeData.event;
+  protected data: TreeItem[] | undefined;
+  constructor(
+    protected readonly context: vscode.ExtensionContext,
+    protected readonly statusBarItem: vscode.StatusBarItem
+  ) {}
+
+  public getTreeItem(
+    element: TreeItem
+  ): vscode.TreeItem | Thenable<vscode.TreeItem> {
+    return element;
+  }
+
+  public getChildren(
+    element?: TreeItem | undefined
+  ): vscode.ProviderResult<TreeItem[]> {
+    if (element === undefined) {
+      return this.data;
+    }
+    return element.children;
+  }
+  protected hideStatusBarItem() {
+    this.statusBarItem.text = constants.extensionName;
+    this.statusBarItem.tooltip = undefined;
+    this.statusBarItem.command = undefined;
+    this.statusBarItem.hide();
+  }
+  protected showStatusBarItem(message: string) {
+    this.statusBarItem.text = constants.refreshingTree;
+    this.statusBarItem.tooltip = message;
+    this.statusBarItem.show();
+  }
+  public refresh(): void {
+    this._onDidChangeTreeData.fire(undefined);
+  }
+
+  public determineTypeLabel(result: CxResult): string | undefined {
+    if (result.label) {
```

Comment:
What will happen if suddenly scs results will have non empty label?

Reply:
If scs result will have  a non empty label so i expect that the vaule need to be the real label so return result.label in that case